packages/codec/lib/types.ts

```diff
@@ -304,9 +304,10 @@ export interface EventDecoding {
   class: Types.ContractType;
   /**
    * The class of the contract that (according to this decoding) defined the event, as a Format.Types.ContractType.
-   * May be omitted if we can't determine it, as may occur in ABI mode.
+   * May be omitted if we can't determine it, as may occur in ABI mode.  If null, this means that it's a file-level
+   * event (which as of right now is just future-proofing).
    */
-  definedIn?: Types.ContractType;
+  definedIn?: Types.ContractType | null;
```

Comment:
Is this a breaking change?

Reply:
Also, does this mean we should include an update to codec-components in this PR, to handle this new `null` case?

Reply:
Ooh, good point, you're absolutely right.  I guess this is technically breaking?  Bleh.  And yeah I'll need to rebase this PR and update components.  Thanks for catching that!

(...is this the entirety of your review or is this just a single early comment?)

Reply:
It was an early comment, but now I think it's also the entirety of the review. We should get this and #6050 into the same release to avoid extra minor bumps

Reply:
Oh hm actually -- it looks like the current component for events doesn't use the `definedIn` field at all.  That'll probably change at some point, but for now, this does not require updating any components after all!